webextension/js/popup.js

```diff
@@ -101,6 +101,28 @@ let Logic = {
     }
     return this._currentIdentity;
   },
+
+  generateIdentityName() {
+    const defaultName = 'Container';
+    let ids = [];
+
+    // This loop populates the 'ids' array with all the already-used ids.
+    this._identities.forEach(identity => {
+      if (identity.name.startsWith(defaultName)) {
+        let id = parseInt(identity.name.substr(defaultName.length))
+        if (id) {
+          ids.push(id);
+        }
+      }
+    });
+
+    // Here we find the first valid id.
+    for (let id = 1;; ++id) {
+      if (ids.indexOf(id) == -1) {
```

Comment:
Also, isn't there a slight possibility of a never-ending loop since the `return` statement is in an `if` block?